plugins/destination/duckdb/client/write.go

```diff
@@ -154,3 +170,74 @@ func (c *Client) WriteTableBatch(ctx context.Context, sc *arrow.Schema, records
 
 	return nil
 }
+
+func transformSchema(sc *arrow.Schema) *arrow.Schema {
+	fields := sc.Fields()
+	for i := range fields {
+		fields[i].Type = transformType(fields[i].Type)
+	}
+	md := sc.Metadata()
+	return arrow.NewSchema(fields, &md)
+}
+
+func transformType(dt arrow.DataType) arrow.DataType {
+	switch {
+	case arrow.TypeEqual(dt, types.ExtensionTypes.UUID) ||
+		arrow.TypeEqual(dt, types.ExtensionTypes.Inet) ||
+		arrow.TypeEqual(dt, types.ExtensionTypes.MAC) ||
+		arrow.TypeEqual(dt, types.ExtensionTypes.JSON) ||
+		dt.ID() == arrow.STRUCT:
+		return arrow.BinaryTypes.String
+	case arrow.IsListLike(dt.ID()):
+		return arrow.ListOf(transformType(dt.(*arrow.ListType).Elem()))
+	default:
+		return dt
+	}
+}
+
+func transformRecord(sc *arrow.Schema, rec arrow.Record) arrow.Record {
+	cols := make([]arrow.Array, rec.NumCols())
+	for i := 0; i < int(rec.NumCols()); i++ {
+		cols[i] = transformArray(rec.Column(i))
+	}
+	return array.NewRecord(sc, cols, rec.NumRows())
+}
+
+func transformArray(arr arrow.Array) arrow.Array {
+	dt := arr.DataType()
+	switch {
+	case arrow.TypeEqual(dt, types.ExtensionTypes.UUID) ||
+		arrow.TypeEqual(dt, types.ExtensionTypes.Inet) ||
+		arrow.TypeEqual(dt, types.ExtensionTypes.MAC) ||
+		arrow.TypeEqual(dt, types.ExtensionTypes.JSON) ||
+		dt.ID() == arrow.STRUCT:
+		return transformToStringArray(arr)
+	case arrow.IsListLike(dt.ID()):
+		child := transformArray(arr.(*array.List).ListValues()).Data()
+		newType := arrow.ListOf(child.DataType())
+		return array.NewListData(array.NewData(newType, arr.Len(), arr.Data().Buffers(), []arrow.ArrayData{child}, arr.NullN(), arr.Data().Offset()))
+	default:
+		return arr
+	}
+}
+
+func transformToStringArray(arr arrow.Array) arrow.Array {
+	bldr := array.NewStringBuilder(memory.DefaultAllocator)
+	for i := 0; i < arr.Len(); i++ {
+		if arr.IsValid(i) {
+			bldr.Append(arr.ValueStr(i))
+		} else {
+			bldr.AppendNull()
+		}
```

Comment:
```suggestion
		if arr.IsNull(i) {
			bldr.AppendNull()
			continue
		}
		bldr.Append(arr.ValueStr(i))
```

Reply:
IsValid is better then IsNull afk as it checks for both.

Reply:
Right but in your suggestion I'll be adding something while it contains an invalid thing so in our conversion if an array has either a null or an invalid thing we will add a null (because there is no such thing as an invalid value in a database - at least in duckdb). this is why both `IsValid` -> maps to Null

Reply:
OK, 1 more time: `IsValid(i) ~ !IsNull(i)`:
* if `IsValid(i)` panics, `IsNull(i)` also panics and vice versa
* `IsValid(i) == true` -> `IsNull(i) == false`
* `IsValid(i) == false` -> `IsNull(i) == true`

So, basically, the code you have:
```go
if ok(i) {
	bldr.Append(arr.ValueStr(i))
} else {
	bldr.AppendNull()
}
```

And what I'm suggesting (early loop exit for less ident):
```go
if !ok(i) {
	bldr.AppendNull()
	continue
}
bldr.Append(arr.ValueStr(i))
```

Reply:
resolving, as it might be considered a nit